test/jdk/jdk/crac/Selector/checkpointLock/Test.java

```diff
@@ -0,0 +1,103 @@
+// Copyright 2019-2021 Azul Systems, Inc.  All Rights Reserved.
+// DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+//
+// This code is free software; you can redistribute it and/or modify it under
+// the terms of the GNU General Public License version 2 only, as published by
+// the Free Software Foundation.
+//
+// This code is distributed in the hope that it will be useful, but WITHOUT ANY
+// WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR
+// A PARTICULAR PURPOSE.  See the GNU General Public License version 2 for more
+// details (a copy is included in the LICENSE file that accompanied this code).
+//
+// You should have received a copy of the GNU General Public License version 2
+// along with this work; if not, write to the Free Software Foundation, Inc.,
+// 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+//
+// Please contact Azul Systems, 385 Moffett Park Drive, Suite 115, Sunnyvale,
+// CA 94089 USA or visit www.azul.com if you need additional information or
+// have any questions.
+
+
+import jdk.crac.*;
+import java.util.Random;
+import java.util.concurrent.atomic.AtomicLong;
+import java.util.stream.LongStream;
+
+
+public class Test implements Resource {
+
+    private static final AtomicLong counter = new AtomicLong(0);
+    private static boolean stop = false;
+    private static final long MIN_TIMEOUT = 100;
+    private static final long MAX_TIMEOUT = 1000;
+
+    private static class TestThread extends Thread {
+        private long timeout;
+
+        TestThread(long timeout) {
+            System.out.println("TestThread " + timeout);
+            this.timeout = timeout;
+        }
+
+        @Override
+        @SuppressWarnings("try")
```

Comment:
Sorry, I missed this initially. What warning does the "try" generate? I.e. will every user will need this?

Reply:
OpenJDK build generates warning : "warning: [try] auto-closeable resource lock is never referenced in body of corresponding try statement" 
I never see this warning compiling tests or apps. It safe to remove from test